### ISSUE TITLE
glibc 2.28 - "Mutex creation failed" error

### ISSUE THREAD

**Issue by asardaes** [RcppCore/RcppParallel#77]
My Linux system now uses glibc 2.28, and my R package (which links to RcppParallel) fails to install/load with the following error:

```
** byte-compile and prepare package for lazy loading
terminate called after throwing an instance of 'std::runtime_error'
  what():  Mutex creation failed
/usr/lib64/R/bin/INSTALL: line 34: 26272 Done                    echo 'tools:::.install_packages()'
     26273 Aborted                 (core dumped) | R_DEFAULT_PACKAGES= LC_COLLATE=C "${R_HOME}/bin/R" $myArgs --slave --args ${args}
```

Based on r-lib/later#63 and [this bug report](https://bugs.archlinux.org/task/59581), I think this may be due to an incompatibility with tinythread and/or TBB. Could you corroborate?

**Comment by eddelbuettel** [comment]
No, we can't as we have no [minimally complete verifiable example](https://stackoverflow.com/help/mcve).

You may also have issues with multiple threading libraries as you suspect.  Maybe you can use RcpParallel in a way that does not require later.

**Comment by eddelbuettel** [comment]
I misread your report. Are you saying it has nothing to do with r-lib/later but that you see the _same symptoms_?

If so, why file it here?  Is this not obviously an Arch Linux issue?  The testing matrix for RcppParallel is [here](https://cloud.r-project.org/web/checks/check_results_RcppParallel.html) and looks fine across several OSs (including two Linux distros).

**Comment by asardaes** [comment]
Yes, I'm seeing the same symptoms. A comment in the Arch Linux bug says:

> Looks like a function name clash. glibc implemented ISO C threads in 2.28 and that mtx_init is being called rather than the R library provided version. Probably worked around with a sed on the R sources. Nothing to be done at the Arch end.

Since r-lib/later had to made adjustments to their src code (and claim the issue is now fixed, see r-lib/later#64), I figured this would also affect RcppParallel and the packages that use it. I may be mistaken though.

**Comment by kevinushey** [comment]
In `later`s the issue was related to a symbol name collision (I think it was once of the `mtx_`-prefixed functions). My understanding is that `later` was expecting to link against a bundled C11 threads library, but unintentionally linked to and used symbols from the system threading library.

AFAIK RcppParallel nor TBB export any symbols of this form, but other collisions could certainly be possible. Ultimately, a stack trace (e.g. try attaching `gdb` to your R process) would help.

**Comment by eddelbuettel** [comment]
Debian isn't yet at glibc 2.28 -- but I am sure we would hear from CRAN where some maintainers are in fact very diligent in checking against forthcoming version.
